Extensions/dnSpy.AsmEditor/Bundle/SaveBundleContentsCommands.cs

```diff
@@ -0,0 +1,85 @@
+	sealed class SaveBundleContentsCommand {
+		[ExportMenuItem(Header = "res:SaveBundleContents", Group = MenuConstants.GROUP_CTX_DOCUMENTS_ASMED_BUNDLE, Order = 0)]
+		sealed class DocumentsCommand : DocumentsContextMenuHandler {
+			public override bool IsVisible(AsmEditorContext context) => SaveBundleContentsCommand.CanExecute(context);
+			public override void Execute(AsmEditorContext context) => SaveBundleContentsCommand.Execute(context);
+		}
+
+		[ExportMenuItem(OwnerGuid = MenuConstants.APP_MENU_EDIT_GUID, Header = "res:SaveBundleContents", Group = MenuConstants.GROUP_APP_MENU_EDIT_ASMED_BUNDLE, Order = 0)]
+		sealed class EditMenuCommand : EditMenuHandler {
+			[ImportingConstructor]
+			public EditMenuCommand(IAppService appService) : base(appService.DocumentTreeView) {
+			}
+			public override bool IsVisible(AsmEditorContext context) => SaveBundleContentsCommand.CanExecute(context);
+			public override void Execute(AsmEditorContext context) => SaveBundleContentsCommand.Execute(context);
+		}
+
+		static bool IsSingleFileBundle(AsmEditorContext context) => context.Nodes.Length == 1 && context.Nodes[0] is BundleDocumentNode;
+		static bool CanExecute(AsmEditorContext context) => SaveBundleContentsCommand.IsSingleFileBundle(context);
+		static void Execute(AsmEditorContext context) {
+			var docNode = context.Nodes[0].GetDocumentNode();
+			var bundleDoc = docNode!.Document as DsBundleDocument;
+			Debug2.Assert(bundleDoc != null);
+			Debug2.Assert(bundleDoc.SingleFileBundle != null);
+			SaveBundle.Save(bundleDoc.SingleFileBundle.Entries.ToArray(), dnSpy_AsmEditor_Resources.SaveBundleContents);
+		}
+	}
+
+	sealed class SaveRawEntryCommand {
+		[ExportMenuItem(Header = "res:SaveRawEntry", Group = MenuConstants.GROUP_CTX_DOCUMENTS_ASMED_BUNDLE, Order = 1)]
+		sealed class DocumentsCommand : DocumentsContextMenuHandler {
+			public override bool IsVisible(AsmEditorContext context) => SaveRawEntryCommand.CanExecute(context);
+			public override void Execute(AsmEditorContext context) => SaveRawEntryCommand.Execute(context);
+		}
+
+		[ExportMenuItem(OwnerGuid = MenuConstants.APP_MENU_EDIT_GUID, Header = "res:SaveRawEntry", Group = MenuConstants.GROUP_APP_MENU_EDIT_ASMED_BUNDLE, Order = 1)]
+		sealed class EditMenuCommand : EditMenuHandler {
+			[ImportingConstructor]
+			public EditMenuCommand(IAppService appService) : base(appService.DocumentTreeView) {
+			}
+			public override bool IsVisible(AsmEditorContext context) => SaveRawEntryCommand.CanExecute(context);
+			public override void Execute(AsmEditorContext context) => SaveRawEntryCommand.Execute(context);
+		}
+
+		static bool IsBundleSingleSelection(AsmEditorContext context) => context.Nodes.Length == 1 && context.Nodes[0] is IBundleEntryNode;
+		static bool CanExecute(AsmEditorContext context) => SaveRawEntryCommand.IsBundleSingleSelection(context);
+		static void Execute(AsmEditorContext context) {
+			var bundleEntryNode = (IBundleEntryNode)context.Nodes[0];
```

Comment:
```suggestion
			var bundleEntryNode = context.Nodes[0] as IBundleEntryNode;
			Debug2.Assert(bundleEntryNode is not null);
```

Use safe cast here and debug assert. Furthermore, please assert that `bundleEntryNode.BundleEntry is not null`